Sources/StreamChatUI/Utils/NukeImageProcessor.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import UIKit
+
+/// This class provides resizing operations for `UIImage`. It internally uses `Nuke` porcessors to implement operations on images.
+open class NukeImageProcessor {
+    /// Crop the image to a given size. The image is center-cropped
+    /// - Parameters:
+    ///   - image: The image to crop
+    ///   - size: The size to which the image needs to be cropped
+    /// - Returns: The cropped image
+    open func crop(image: UIImage, to size: CGSize) -> UIImage? {
+        let imageProccessor = ImageProcessors.Resize(size: size, crop: true)
```

Comment:
`ImageProcessors.Resize(size: size, crop: true)`
should be dependency-injected with protocol usage